force-app/main/schema/classes/collaborationgroup/ChatterGroupMapper.cls

```diff
@@ -0,0 +1,114 @@
+ * @group-content ../../ApexDocContent/
+ * @description Mapper class to retrieve data needed for User Models
+ */
+public virtual with sharing class ChatterGroupMapper {
+    /*********************************************
+     * @description Instance for Singleton Pattern
+     **********************************************/
+    @TestVisible
+    private static ChatterGroupMapper instance;
+
+    /*********************************************************************
+     * @description Empty constructor is not public for Singleton Pattern
+     **********************************************************************/
+    protected ChatterGroupMapper() {
+    }
+
+    /*****************************************************************************
+     * @description Static method to get the current instance for Singleton pattern
+     * @return The instance of ChatterGroupMapper.
+     ******************************************************************************/
+    public static ChatterGroupMapper getInstance() {
+        if (instance == null) {
+            instance = new ChatterGroupMapper();
+        }
+
+        return instance;
+    }
+
+    /************************************************************************************************************
+     * @description Retrieve a ChatterGroupModel representing a chatter group and its basic fields
+     * @param chatterGroupId The Id of the chatter group to retrieve
+     * @return A ChatterGroupModel
+     ************************************************************************************************************/
+    public virtual ChatterGroupModel getChatterGroupModelById(Id chatterGroupId) {
+        CollaborationGroup chatterGroupRetrieved = [
+            SELECT Id, Name, Description, SmallPhotoUrl
+            FROM CollaborationGroup
+            WHERE Id = :chatterGroupId
+            WITH SECURITY_ENFORCED
+        ];
+
+        return new ChatterGroupModel(
+            chatterGroupRetrieved.Id,
+            chatterGroupRetrieved.Name,
+            chatterGroupRetrieved.Description,
+            chatterGroupRetrieved.SmallPhotoUrl
+        );
+    }
+
+    /************************************************************************************************************
+     * @description Retrieve a list of ChatterGroupModel representing a chatter group and their basic fields
+     * @param chatterGroupNameMatch The string to match chatter group names to
+     * @param recordLimit The number of records to limit the query to
+     * @return A list of Chatter Group Models with names that match the search string
+     ************************************************************************************************************/
+    public virtual List<ChatterGroupModel> getChatterGroupModelsLikeName(
+        String chatterGroupNameMatch,
+        Integer recordLimit
+    ) {
+        String sanitizedNameMatchSearch = '%' + String.escapeSingleQuotes(chatterGroupNameMatch) + '%';
```

Comment:
Not a big fan of this variable name.

Reply:
Changed to escapedChatterGroupNameMatch. That is what the string is, a match for a name of an object escaped. Hopefully this helps